services/controllerhost/controllerhost.go

```diff
@@ -1251,3 +1279,97 @@ func (mcp *Mcp) CreateRemoteZoneConsumerGroupExtent(ctx thrift.Context, createRe
 	lclLg.Info("CreateRemoteZoneConsumerGroupExtent: Extent added to consumer group")
 	return nil
 }
+
+// ValidateDestZoneConfigUpdateRequest validates the zone configs for a UpdateDestinationRequest
+func (mcp *Mcp) ValidateDestZoneConfigUpdateRequest(ctx thrift.Context, logger bark.Logger, updateRequest *shared.UpdateDestinationRequest) (bool, error) {
+	// first read the destination
+	destDesc, err := mcp.mClient.ReadDestination(ctx, &shared.ReadDestinationRequest{
+		DestinationUUID: common.StringPtr(updateRequest.GetDestinationUUID()),
+	})
+	if err != nil {
+		logger.WithField(common.TagErr, err).Error("ValidateDestZoneConfigUpdateRequest: ReadDestination in local failed")
+		return false, err
+	}
+
+	destPath := destDesc.GetPath()
+
+	localReplicator, err := mcp.GetClientFactory().GetReplicatorClient()
+	if err != nil {
+		logger.WithField(common.TagErr, err).Error("ValidateDestZoneConfigUpdateRequest: GetReplicatorClient failed")
+		return false, err
+	}
+
+	for _, zoneConfig := range updateRequest.GetZoneConfigs() {
+		if strings.EqualFold(zoneConfig.GetZone(), mcp.context.localZone) {
+			continue
+		}
+
+		remoteDest, err := localReplicator.ReadDestinationInRemoteZone(ctx, &shared.ReadDestinationInRemoteZoneRequest{
+			Zone: common.StringPtr(zoneConfig.GetZone()),
+			Request: &shared.ReadDestinationRequest{
+				Path: common.StringPtr(destPath),
+			},
+		})
+		if err == nil {
+			// path exists in remote. If uuid is different, then fail the validation
+			if remoteDest.GetDestinationUUID() != updateRequest.GetDestinationUUID() {
+				logger.WithField(common.TagZoneName, common.FmtZoneName(zoneConfig.GetZone())).Error(`destination exists in remote but UUID is different`)
+				return false, nil
+			}
+		} else if _, ok := err.(*shared.EntityNotExistsError); ok {
+			// path doesn't exist in remote, validation succeeds
+			continue
+		} else {
+			// some other error
+			return false, err
+		}
+	}
+	return true, nil
+}
+
+// ValidateCgZoneConfigUpdateRequest validates the zone configs for a UpdateConsumerGroupRequest
+func (mcp *Mcp) ValidateCgZoneConfigUpdateRequest(ctx thrift.Context, logger bark.Logger, updateRequest *shared.UpdateConsumerGroupRequest) (bool, error) {
```

Comment:
as with 'ValidateDestZoneConfigUpdateRequest', this could get called from the CreateDestination path too, right?